R/parse-datetime.r

```diff
@@ -11,30 +11,32 @@
 
 #' @param date contains the date to be formatted
 #' @param mode gives the desired output format (see above)
-datetime_fn <- function(createdat, mode="since", permissive=FALSE) {
+#' @param permissive logical. If false, will error with an improper type.  If true, will just warn.
+datetime_fn <- function(createdat, mode = "since", permissive = FALSE) {
   
   # track which records are NA before the mungebit
   originally_NA <- is.na(createdat)
   
   # do the conversion
-  datetime <- suppressWarnings(lubridate:::ymd_hms(createdat))
+  datetime <- suppressWarnings(lubridate::ymd_hms(createdat))
   if (any(is.na(datetime))) {
     if (mode == 'hod') { stop('Cannot extract hour from date-only.') }
-    datetime <- suppressWarnings(lubridate:::ymd(createdat))
+    datetime <- suppressWarnings(lubridate::ymd(createdat))
   }
   
   # check to see if any records are converted to NA
-  if (any(is.na(datetime[!originally_NA])) && !isTRUE(permissive)) { 
-    stop('Improper date format.') 
+  if (any(is.na(datetime[!originally_NA]))) {
+    error_method <- if (isTRUE(permissive)) warning else stop
+    error_method('Improper date format.')
   }
 
   suppressWarnings(switch(mode,
     "since" = as.numeric(as.Date(datetime)),
-    "hod" = lubridate:::hour(datetime),
+    "hod" = lubridate::hour(datetime),
     "dow" = weekdays(datetime),
-    "dom" = lubridate:::day(datetime),
+    "dom" = lubridate::day(datetime),
     "doy" = syberiaMungebits:::get.doy(datetime),
-    "moy" = lubridate:::month(datetime),
+    "moy" = lubridate::month(datetime),
     "holiday" = syberiaMungebits:::is.holiday(datetime),
```

Comment:
Not that it matters much, but you could get a free performance increase if you did:

```r
(function(datetime) {
switch(mode, ...)
})(as.Date(datetime))
```

and then modify the functions below to assume it's already been `as.Date`d to prevent calculating it multiple times.